_lib/wordpress_office_processor.py

```diff
@@ -18,57 +19,92 @@ def posts_at_url(url):
-            top_story[attr] = custom_fields[attr][0]
+            new_attr = attr.replace('top_story_', '')
+            top_story[new_attr] = custom_fields[attr]
 
     # convert top story links into a proper list
-    top_story_links = []
-    for x in xrange(0, 5):
-        key = 'top_story_links_%s_link' % x
-        if key in custom_fields:
-            top_story_links.append(custom_fields[key])
-
-    if top_story_links:
-        top_story['top_story_links'] = top_story_links
+    if 'top_story_links' in custom_fields:
+        top_story['links'] = custom_fields['top_story_links']
+    else:
+        top_story_links = []
+        for x in range(5):
+            key = 'top_story_links_%s' % x
+            if key in custom_fields:
+                top_story_links.append({'url': custom_fields[key][0],
+                                        'label': custom_fields[key][1]})
+
+        if top_story_links:
+            top_story['links'] = top_story_links
 
     if top_story:
-        item['top_story'] = top_story
+        post['top_story'] = top_story
 
     # create list of office resource dicts
-    item['resources'] = []
-    for x in xrange(0, 4):
-        resource = {}
-        fields = ['title', 'desc', 'icon', 'link']
-        for field in fields:
-            field_name = 'resource_%s_%s' % (str(x), field)
-            if field_name in custom_fields and custom_fields[field_name][0] != '':
-                if field == 'link':
-                    resource[field] = custom_fields[field_name]
-                else:
-                    resource[field] = custom_fields[field_name][0]
-
-        if resource:
-            item['resources'].append(resource)
-
-    del item['custom_fields']
-
-    return item
+    if 'resources' in custom_fields:
+        post['resources'] = custom_fields['resources']
+    else:
+        post['resources'] = []
+        for x in range(4):
+            resource = {}
+            fields = ['title', 'desc', 'icon', 'link']
+            for field in fields:
+                field_name = 'resource_%s_%s' % (str(x), field)
+                if field_name in custom_fields and custom_fields[field_name]:
+                    if field == 'link':
+                        resource[field] = \
+                            {'url': custom_fields[field_name][0],
+                             'label': custom_fields[field_name][1]}
+                    else:
+                        resource[field] = custom_fields[field_name]
+
+            if resource:
+                post['resources'].append(resource)
+
+    # add other custom fields
+    names = ['og_title', 'og_image', 'og_desc', 'twtr_text', 'twtr_lang',
+             'twtr_rel', 'twtr_hash', 'utm_campaign', 'utm_term',
+             'utm_content', 'short_title']
+    for name in names:
+        if name in custom_fields:
+            post[name] = custom_fields[name]
+
+    for related in ['related_hero', 'related_contact', 'related_faq']:
+        if related in custom_fields:
```

Comment:
Ah, gotcha. Thanks for showing me that style, though in this case we don't really have any reason to add any more conditions at that level to warrant such a style change. In other terms, we only care about data when it exists within `custom_fields` so it wouldn't make sense to account for the other case. And even if it were the case, we would probably want to execute more code instead of just returning/continuing.